code/android-signal-library/src/main/java/com/adobe/marketing/mobile/signal/SignalHitProcessor.kt

```diff
@@ -0,0 +1,117 @@
+/*
+  Copyright 2022 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+package com.adobe.marketing.mobile.signal
+
+import androidx.annotation.VisibleForTesting
+import com.adobe.marketing.mobile.services.*
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+internal class SignalHitProcessor : HitProcessing {
+    private val networkService: Networking
```

Comment:
We allow overriding network service. Let's make it an optional and add null checks till we find a better way of handling such use cases.

Reply:
https://github.com/adobe/aepsdk-core-android/blob/258d31272a6dd41f3fdacd2c6045297f93db6a64/code/android-core-library/src/phone/java/com/adobe/marketing/mobile/services/ServiceProvider.java#L124
The `ServiceProvide` has done the null check